src/cluster/Telemetry.cc

```diff
@@ -131,28 +127,41 @@ void VerboseTelemetry::OnIncomingEvent(std::string_view topic, std::string_view
 
 namespace {
 
-std::string determine_script_location() {
-    std::string result = "none";
-
-    if ( zeek::detail::call_stack.empty() )
-        return result;
+// Cache CallExpr pointers to their location.
+std::map<const zeek::detail::CallExpr*, std::string> location_cache;
```

Comment:
I'd just move this into `determine_script_location` as a `static` which should behave identical, but would be easier to follow.

I am not know whether `CallExpr*` are equivalent to script locations, but I assume you checked.